src/app/child-dev-project/children/children-list/children-list.component.ts

```diff
@@ -228,24 +216,15 @@ export class ChildrenListComponent implements OnInit, AfterViewInit {
     }
   }
 
-  private loadData(replaceUrl: boolean = false) {
-    this.childrenService
-      .getChildren()
-      .pipe(untilDestroyed(this))
-      .subscribe((children) => {
-        this.childrenList = children;
-        const centers = children
-          .map((c) => c.center)
-          .filter((value, index, arr) => value && arr.indexOf(value) === index);
-        this.centerFS.initOptions(centers, "center");
-
-        this.applyFilterSelections(replaceUrl);
-      });
-    this.childrenService
-      .getAttendances()
-      .pipe(untilDestroyed(this))
-      .subscribe((results) => this.prepareAttendanceData(results));
+  private async loadData(replaceUrl: boolean = false) {
+    this.childrenList = await this.entityMapperService.loadType<Child>(Child);
```

Comment:
The point of using `ChildrenService` is to let that load the school relation or other related data. Why should this be changed?

Reply:
This makes it independent of the entity that is being displayed. I got rid of the `ChildrenService` and only use the `EntityMapper`. This means data is only loaded once it is actually required.

Reply:
I think it is time to agree on an architecture which will always be present, e.g. there is always a child entity which stands in a relation to schools/team and we use this to optimize the application to this scenario or we want even this behavior to be configurable and every entity can stand in a relation to any other.